app/models/va_facility.rb

```diff
@@ -49,29 +46,25 @@ def self.find_by(id:)
     prefix, station = id.split('_')
     adapter = client_adapter(ID_PREFIXES[prefix])
     return nil unless adapter
-    request = adapter.find_by(id: station)
-    responses = multi_client.run([request])
-    adapter.class&.from_gis(responses.first.first) unless responses.first.blank?
+    adapter.get(station)
   end
 
   def self.service_whitelist(prefix)
-    client_adapter(prefix)&.service_whitelist
-  end
-
-  def self.multi_client
-    @mc ||= Facilities::MultiClient.new
+    client_adapter(prefix)&.adapter&.service_whitelist
   end
 
   def self.client_adapter(prefix)
-    @client_map ||= init_adapters
+    @lock.synchronize do
+      @client_map ||= init_adapters
+    end
     @client_map[prefix]
   end
 
   def self.init_adapters
     {
-      HEALTH => VHAFacilityAdapter.new,
-      CEMETERY => NCAFacilityAdapter.new,
-      BENEFITS => VBAFacilityAdapter.new
+      HEALTH => Facilities::LocalStore.new(ENV['VHA_MAPSERVER_URL'], nil, VHAFacilityAdapter),
+      CEMETERY => Facilities::LocalStore.new(ENV['NCA_MAPSERVER_URL'], nil, NCAFacilityAdapter),
```

Comment:
I'll consider it. I felt there was testability benefit in keeping the json -> model conversion logic in a simple adapter class that had no request logic, no need for stubbing, etc., so I could blast out a bunch of data format/conversion tests for the noisy data we are getting for these facilities.